lib/TPP/Transforms/CombineXsmmPass.cpp

```diff
@@ -126,19 +126,31 @@ struct CombineXsmmOp : public OpRewritePattern<xsmm::BrgemmOp> {
 
     // Replace and delete the old invokes and their dispatches
     rewriter.create<xsmm::FusedBrgemmOp>(loc, dtype, invokeOperands);
+    assert(brgemmOp.use_empty());
     rewriter.eraseOp(brgemmOp);
-    rewriter.eraseOp(brgemmOp.getOperand(0).getDefiningOp());
+    if (brgemmOp.getOperand(0).getDefiningOp()->use_empty()) {
```

Comment:
Please cache `brgemmOp.getOperand(0)` and the following repeated pattern below for clarity.